src/miner.cpp

```diff
@@ -728,6 +732,14 @@ namespace pos {
 
         auto pblock = std::make_shared<CBlock>(pblocktemplate->block);
 
+        // Set search time if null or last block has changed
+        if (!nLastCoinStakeSearchTime || lastBlockSeen != tip->GetBlockHash()) {
+            // Expect this to be called sequentially but let's lock to be safe.
+            CLockFreeGuard lock(Staker::cs_staker);
```

Comment:
Now is single thread mining, do we still need lock?

Reply:
No, but I'm paranoid that someone will make it "concurrent" again. I will remove.